utils/timeofday.go

```diff
@@ -0,0 +1,67 @@
+package utils
+
+import (
+	"time"
+)
+
+// TimeOfDay represents a local time of day value
+type TimeOfDay struct {
+	Hour   int
+	Minute int
+	Second int
+	Nanos  int
+}
+
+// NewTimeOfDay creates a new time of day
+func NewTimeOfDay(hour, minute, second, nanos int) TimeOfDay {
+	return TimeOfDay{Hour: hour, Minute: minute, Second: second, Nanos: nanos}
+}
+
+// ExtractTimeOfDay extracts the time of day from the give datetime
+func ExtractTimeOfDay(dt time.Time) TimeOfDay {
+	return NewTimeOfDay(dt.Hour(), dt.Minute(), dt.Second(), dt.Nanosecond())
+}
+
+// Equals determines equality for this type
+func (t TimeOfDay) Equals(other TimeOfDay) bool {
+	return t.Hour == other.Hour && t.Minute == other.Minute && t.Second == other.Second && t.Nanos == other.Nanos
+}
+
+// Compare compares this time of day to another
+func (t TimeOfDay) Compare(other TimeOfDay) int {
+	if t.Hour != other.Hour {
+		return t.Hour - other.Hour
+	}
+	if t.Minute != other.Minute {
+		return t.Minute - other.Minute
+	}
+	if t.Second != other.Second {
+		return t.Second - other.Second
+	}
+	return t.Nanos - other.Nanos
+}
+
+// Format formats this time of day as a string
+func (t TimeOfDay) Format(layout string) string {
+	// upgrade us to a date time so we can use standard time.Time formatting
+	d := time.Date(1970, 1, 1, t.Hour, t.Minute, t.Second, t.Nanos, time.UTC)
+	return d.Format(layout)
+}
+
+// String returns the ISO8601 representation
+func (t TimeOfDay) String() string {
+	return t.Format(iso8601Time)
```

Comment:
Doesn't do anything clever with dropping nanoseconds if they're zero etc - but instead is consistent with formatting of time in datetimes.

Reply:
I'm a bit sad about this but I suppose its consistent with our treatment of dates / locations.

Reply:
well not saying we shouldn't change it - just that times and dates should use the same formatting

Reply:
We are definitely training people in the new world that they need to use `format_` to get pretty things, so I think its consistent / fine.

For the record elastic doesn't have time support, so we would need to use a normalized time value for searching anyways, but really all that requires is zero padding the hour to guarantee lexical ordering. (IE, it doesn't really affect this decision either way, though from a purity point of view I agree having all our parsed times looks exactly the same sure feels right)